core/src/main/java/org/apache/carbondata/core/scan/result/BlockletScannedResult.java

```diff
@@ -663,6 +663,12 @@ public boolean hasNext() {
       return true;
     } else if (pageCounter < pageFilteredRowCount.length) {
       pageCounter++;
+      if(pageCounter >= pageFilteredRowCount.length) {
```

Comment:
fillDataChunks calls freeDataChunkMemory(), but your changes skip this operation, i think "filteredRowCount == 0" has no problem, but the last page need call this method to free memory